src-graphql/client.js

```diff
@@ -24,11 +24,11 @@ function fetchAllProductResources(product, client) {
   const images = product.images;
   const variants = product.variants;
 
-  if (images && images[images.length - 1].hasNextPage.valueOf()) {
+  if (images && images.length && images[images.length - 1].hasNextPage.valueOf()) {
```

Comment:
Yeah, truthy and falsey tests don't actually attempt to coerce the value to a boolean. I'm reading up on it, but boxed primitives work mostly as you'd expect until you deal with booleans. I'm trying to think of other ways to do this without boxing everything. If you have any suggestions around this, please let me know.